src/jit/compiler.hpp

```diff
@@ -2768,7 +2769,28 @@ inline bool Compiler::fgIsThrowHlpBlk(BasicBlock* block)
         return false;
     }
 
-    GenTreePtr call = block->bbTreeList->gtStmt.gtStmtExpr;
+    GenTree* call = nullptr;
+    if (block->IsLIR())
+    {
+        LIR::Range& blockRange = LIR::AsRange(block);
+        call                   = blockRange.LastNode();
```

Comment:
It would be good to have a generalized block->GetLastNode() method, that would work in the LIR or non-LIR case.